worker/raft/rafttransport/worker_test.go

```diff
@@ -230,12 +231,28 @@ func (s *WorkerSuite) TestTransportWorkerStopped(c *gc.C) {
 
 func (s *WorkerSuite) TestTransportTimeout(c *gc.C) {
 	config := s.config
-	config.Timeout = time.Millisecond
+	config.Timeout = time.Nanosecond
 	worker := s.newWorker(c, config)
 
-	_, err := s.requestVote(worker)
-	c.Assert(err, gc.ErrorMatches, "dial failed:.*timed out.*")
+	rv := func() error {
+		_, err := s.requestVote(worker)
+		return err
+	}
+
+	// This test previously had intermittent failures with a worker timeout
+	// set to 1 millisecond. With the nanosecond timeout, this should always
+	// break out first time, but it remains as a fail-safe.
+	deadline := time.After(coretesting.LongWait)
+	var err error
+	for err = rv(); !strings.Contains(err.Error(), "dial failed"); err = rv() {
```

Comment:
I"m pretty sure (err == nil) will panic if you try to call err.Error().
Though if I set the timeout to 1s, then I get:
read tcp 127.0.0.1:59120->127.0.0.1:33506: i/o timeout

So presumably we always get an error. However, we should definitely be careful about getting a 'nil' back if we ever change how the worker acts.

I don't think you need to explain how the test 'used to work' in the comment. But having a good explanation of now is good.

// Watch out for an intermittent failure where the dial succeeds before we manage to fail the dial.